Test failure looks like a real thing:
```
I1203 08:30:12.853367   22739 factory.go:173] creating scheduler with fit predicates 'map[NoDiskConflict:{} MatchNodeSelector:{} HostName:{} PodFitsHostPorts:{} PodFitsResources:{}]' and priority functions 'map[LeastRequestedPriority:{} BalancedResourceAllocation:{} SelectorSpreadPriority:{}]
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x28 pc=0x745de1]

goroutine 55 [running]:
k8s.io/kubernetes/plugin/pkg/scheduler.func·002()
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/plugin/pkg/scheduler/scheduler.go:111 +0x41
k8s.io/kubernetes/pkg/util.func·006()
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:126 +0x61
k8s.io/kubernetes/pkg/util.Until(0xc2088fc1a0, 0x3b9aca00, 0xc20804c480)
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:127 +0x8f
k8s.io/kubernetes/pkg/util.Forever(0xc2088fc1a0, 0x3b9aca00)
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/util.go:109 +0x3e
created by k8s.io/kubernetes/plugin/pkg/scheduler.(*Scheduler).Run
	/workspace/kubernetes/_output/local/go/src/k8s.io/kubernetes/plugin/pkg/scheduler/scheduler.go:113 +0xa7
```

--- a/Godeps/Godeps.json
+++ b/Godeps/Godeps.json
@@ -549,7 +549,7 @@
 		},
 		{
 			"ImportPath": "github.com/juju/ratelimit",
-			"Rev": "772f5c38e468398c4511514f4f6aa9a4185bc0a0"
+			"Rev": "77ed1c8a01217656d2080ad51981f6e99adaa177"
 		},
 		{
 			"ImportPath": "github.com/kardianos/osext",

--- a/Godeps/_workspace/src/github.com/juju/ratelimit/ratelimit.go
+++ b/Godeps/_workspace/src/github.com/juju/ratelimit/ratelimit.go
@@ -8,10 +8,10 @@
 package ratelimit
 
 import (
+	"math"
 	"strconv"
 	"sync"
 	"time"
-	"math"
 )
 
 // Bucket represents a token bucket that fills at a predetermined rate.
@@ -169,6 +169,30 @@ func (tb *Bucket) takeAvailable(now time.Time, count int64) int64 {
 	}
 	tb.avail -= count
 	return count
+}
+
+// Available returns the number of available tokens. It will be negative
+// when there are consumers waiting for tokens. Note that if this
+// returns greater than zero, it does not guarantee that calls that take
+// tokens from the buffer will succeed, as the number of available
+// tokens could have changed in the meantime. This method is intended
+// primarily for metrics reporting and debugging.
+func (tb *Bucket) Available() int64 {
+	return tb.available(time.Now())
+}
+
+// available is the internal version of available - it takes the current time as
+// an argument to enable easy testing.
+func (tb *Bucket) available(now time.Time) int64 {
+	tb.mu.Lock()
+	defer tb.mu.Unlock()
+	tb.adjust(now)
+	return tb.avail
+}
+
+// Capacity returns the capacity that the bucket was created with.
+func (tb *Bucket) Capacity() int64 {
+	return tb.capacity
 }
 
 // Rate returns the fill rate of the bucket, in tokens per second.

--- a/pkg/util/throttle.go
+++ b/pkg/util/throttle.go
@@ -28,7 +28,7 @@ type RateLimiter interface {
 	Stop()
 }
 
-type tickRateLimiter struct {
+type tokenBucketRateLimiter struct {
 	limiter *ratelimit.Bucket
 }
 
@@ -39,7 +39,7 @@ type tickRateLimiter struct {
 // The maximum number of tokens in the bucket is capped at 'burst'.
 func NewTokenBucketRateLimiter(qps float32, burst int) RateLimiter {
 	limiter := ratelimit.NewBucketWithRate(float64(qps), int64(burst))
-	return &tickRateLimiter{limiter}
+	return &tokenBucketRateLimiter{limiter}
 }
 
 type fakeRateLimiter struct{}
@@ -48,16 +48,16 @@ func NewFakeRateLimiter() RateLimiter {
 	return &fakeRateLimiter{}
 }
 
-func (t *tickRateLimiter) TryAccept() bool {
+func (t *tokenBucketRateLimiter) TryAccept() bool {
 	return t.limiter.TakeAvailable(1) == 1
 }
 
 // Accept will block until a token becomes available
-func (t *tickRateLimiter) Accept() {
+func (t *tokenBucketRateLimiter) Accept() {
 	t.limiter.Wait(1)
 }
 
-func (t *tickRateLimiter) Stop() {
+func (t *tokenBucketRateLimiter) Stop() {
 }
 
 func (t *fakeRateLimiter) TryAccept() bool {

--- a/pkg/util/throttle.go
+++ b/pkg/util/throttle.go
@@ -26,6 +26,11 @@ type RateLimiter interface {
 	Accept()
 	// Stop stops the rate limiter, subsequent calls to CanAccept will return false
 	Stop()
+	// Saturation returns a percentage number which describes how saturated
+	// this rate limiter is.
+	// Usually we use token bucket rate limiter. In that case,
+	// 1.0 means no tokens are available; 0.0 means we have a full bucket of tokens to use.
+	Saturation() float64
 }
 
 type tokenBucketRateLimiter struct {
@@ -52,6 +57,12 @@ func (t *tokenBucketRateLimiter) TryAccept() bool {
 	return t.limiter.TakeAvailable(1) == 1
 }
 
+func (t *tokenBucketRateLimiter) Saturation() float64 {
+	capacity := t.limiter.Capacity()
+	avail := t.limiter.Available()
+	return float64(capacity-avail) / float64(capacity)
+}
+
 // Accept will block until a token becomes available
 func (t *tokenBucketRateLimiter) Accept() {
 	t.limiter.Wait(1)
@@ -62,6 +73,10 @@ func (t *tokenBucketRateLimiter) Stop() {
 
 func (t *fakeRateLimiter) TryAccept() bool {
 	return true
+}
+
+func (t *fakeRateLimiter) Saturation() float64 {
+	return 0
 }
 
 func (t *fakeRateLimiter) Stop() {}

--- a/pkg/util/throttle_test.go
+++ b/pkg/util/throttle_test.go
@@ -17,6 +17,7 @@ limitations under the License.
 package util
 
 import (
+	"math"
 	"testing"
 	"time"
 )
@@ -61,5 +62,28 @@ func TestThrottle(t *testing.T) {
 	}
 	if time.Now().Before(expectedFinish) {
 		t.Error("rate limit was not respected, finished too early")
+	}
+}
+
+func TestRateLimiterSaturation(t *testing.T) {
+	const e = 0.000001
+	tests := []struct {
+		capacity int
+		take     int
+
+		expectedSaturation float64
+	}{
+		{1, 1, 1},
+		{10, 3, 0.3},
+	}
+	for i, tt := range tests {
+		rl := NewTokenBucketRateLimiter(1, tt.capacity)
+		for i := 0; i < tt.take; i++ {
+			rl.Accept()
+		}
+		if math.Abs(rl.Saturation()-tt.expectedSaturation) > e {
+			t.Fatalf("#%d: Saturation rate difference isn't within tolerable range\n want=%f, get=%f",
+				i, tt.expectedSaturation, rl.Saturation())
+		}
 	}
 }

--- a/plugin/pkg/scheduler/scheduler_test.go
+++ b/plugin/pkg/scheduler/scheduler_test.go
@@ -307,6 +307,10 @@ func (fr *FakeRateLimiter) TryAccept() bool {
 	return true
 }
 
+func (fr *FakeRateLimiter) Saturation() float64 {
+	return 0
+}
+
 func (fr *FakeRateLimiter) Stop() {}
 
 func (fr *FakeRateLimiter) Accept() {

